extensions/binding/org.eclipse.smarthome.binding.dmx/src/main/java/org/eclipse/smarthome/binding/dmx/handler/DimmerThingHandler.java

```diff
@@ -0,0 +1,328 @@
+                    ValueSet targetValueSet = ((OnOffType) command).equals(OnOffType.ON) ? turnOnValue : turnOffValue;
+                    IntStream.range(0, channels.size()).forEach(i -> {
+                        channels.get(i).setChannelAction(
+                                new FadeAction(fadeTime, channels.get(i).getValue(), targetValueSet.getValue(i), -1));
+                    });
+                } else if (command instanceof HSBType) {
+                    logger.trace("adding color fade to channels in thing {}", this.thing.getUID());
+                    HSBType thisCommand = (HSBType) command;
+                    ValueSet rgbValue = new ValueSet(fadeTime, -1);
+                    rgbValue.addValue(thisCommand.getRed());
+                    rgbValue.addValue(thisCommand.getGreen());
+                    rgbValue.addValue(thisCommand.getBlue());
+                    IntStream.range(0, channels.size()).forEach(i -> {
+                        channels.get(i).setChannelAction(new FadeAction(fadeTime, rgbValue.getValue(i), -1));
+                    });
+                } else if (command instanceof RefreshType) {
+                    logger.trace("sending update on refresh to channel {}:color", this.thing.getUID());
+                    updateState(channelUID, UnDefType.NULL);
+                } else {
+                    logger.debug("command {} not supported in channel {}:color", command.getClass(),
+                            this.thing.getUID());
+                }
+                break;
+            case CHANNEL_SWITCH:
+                if (command instanceof OnOffType) {
+                    logger.trace("adding {} fade to channels in thing {}", command, this.thing.getUID());
+                    ValueSet targetValueSet = ((OnOffType) command).equals(OnOffType.ON) ? turnOnValue : turnOffValue;
+                    IntStream.range(0, channels.size()).forEach(i -> {
+                        channels.get(i).setChannelAction(
+                                new FadeAction(fadeTime, channels.get(i).getValue(), targetValueSet.getValue(i), -1));
+                    });
+                } else if (command instanceof RefreshType) {
+                    logger.trace("sending update on refresh to channel {}:switch", this.thing.getUID());
+                    updateState(channelUID, channels.get(0).getValue() > 0 ? OnOffType.ON : OnOffType.OFF);
+                } else {
+                    logger.debug("command {} not supported in channel {}:switch", command.getClass(),
+                            this.thing.getUID());
+                }
+                break;
+            default:
+                logger.debug("channel {} not supported in thing {}", channelUID.getId(), this.thing.getUID());
+        }
+
+    }
+
+    protected void checkColorChannel() {
+        // check if we need to update the channel configuration and update again if necessary
```

Comment:
@kaikreuzer this method adds or removes the color channel, depending on the DMX channel configuration. It is called once after each call to initalize().